app/parse.py

```diff
@@ -15,8 +28,62 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product: Tag) -> Product:
+    product = Product(
+        title=product.select_one(".title")["title"],
+        description=product.select_one(".description").text.replace("\xa0", " "),
+        price=float(product.select_one(".price").text.replace("$", "")),
+        rating=len(product.select(".ws-icon-star")),
+        num_of_reviews=int(product.select_one(".review-count").text.split()[0]),
+    )
+    return product
+
+
+def get_products_from_page(url: str) -> list[Product]:
+    print(f"getting products from {url} via get products from page")
+    response = requests.get(url)
+    print(f"Status code {response.status_code}")
+    soup = BeautifulSoup(response.text, "html.parser")
+    products = soup.find_all("div", {"class": "card-body"})
+
+    if not products:
+        print(f"No products found via static scraping. "
+              f"Trying dynamic scraping for {url}")
+        dynamic_html = get_dynamic_content(url)
```

Comment:
Ensure that the `get_dynamic_content` function returns the expected HTML structure. If the structure differs from the static content, the parsing logic might fail. Consider adding error handling or logging to manage unexpected HTML structures.